Add recipe for consult-dir

karthink opened the pull request:
### Brief summary of what the package does

Consult-dir implements commands to jump to/switch directories easily in Emacs, including when using other Emacs commands.

The user can switch to any directory they've visited recently, or to a project or bookmarked directory. They can do this at any time, including when using the minibuffer. The minibuffer prompt is then replaced with the chosen directory.

The readme has demos showing how this works and some applications.

There is a discussion/review of this package [here](https://github.com/minad/consult/issues/373) with @minad, the author of the Consult package which this builds on.

### Direct link to the package repository

https://github.com/karthink/consult-dir

### Your association with the package

I am the maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this -- just a very quick pass over the checklist:


byte-compile (using Emacs 27.1):
```
consult-dir.el:147:13:Warning: reference to free variable `project--list'
In end of data:
consult-dir.el:300:1:Warning: the function `project--read-project-list' is not
    known to be defined.
```
For what it's worth, I don't see either of these private symbols defined by project.el, so something more than `(require 'project)` seems to be needed here.

checkdoc (using version 0.6.2):
- No issues!

package-lint (using version 20210528.2348):
- No issues!

Other [lints](https://github.com/riscy/melpazoid):
- consult-dir.el#L291: `concat` may be unneeded; `insert` can take multiple arguments
- Add a LICENSE file that GitHub can detect (e.g. no markup) if possible

karthink commented:
Fixed in the latest commit.

> For what it's worth, I don't see either of these private symbols defined by project.el, so something more than `(require 'project)` seems to be needed here.

I needed project 0.6.0 or higher, added it to the requirements. It now byte compiles without warnings.

> Other [lints](https://github.com/riscy/melpazoid):
> - consult-dir.el#L291: `concat` may be unneeded; `insert` can take multiple arguments
> - Add a LICENSE file that GitHub can detect (e.g. no markup) if possible

Addressed both, added GPL V3 license.

riscy commented:
Thanks!
